lib/new_relic/agent/transaction/abstract_segment.rb

```diff
@@ -327,6 +331,25 @@ def transaction_state
             Tracer.state
           end
         end
+
+        def invoke_callback
```

Comment:
TODO: source comment explaining the lack of a public API and performance concerns given that a proc could do anything.

Reply:
Addressed with [dfb892b](https://github.com/newrelic/newrelic-ruby-agent/pull/2178/commits/dfb892bff282a7be80c6c94fc999318a9405e77c)